provider/queue.go

```diff
@@ -142,40 +142,52 @@ func (q *Queue) work() {
 }
 
 func (q *Queue) queueKey(id uint64) datastore.Key {
-	return datastore.NewKey(strconv.FormatUint(id, 10))
+	s := fmt.Sprintf("%016X", id)
+	return datastore.NewKey(s)
 }
 
-// crawl over the queue entries to find the head and tail
-func getQueueHeadTail(ctx context.Context, name string, datastore datastore.Datastore) (uint64, uint64, error) {
-	q := query.Query{}
-	results, err := datastore.Query(q)
+func getQueueHeadTail(ctx context.Context, datastore datastore.Datastore) (uint64, uint64, error) {
+	head, err := getQueueHead(datastore)
 	if err != nil {
 		return 0, 0, err
 	}
+	tail, err := getQueueTail(datastore)
+	if err != nil {
+		return 0, 0, err
+	}
+	return head, tail, nil
+}
 
-	var tail uint64
-	var head uint64 = math.MaxUint64
-	for entry := range results.Next() {
-		trimmed := strings.TrimPrefix(entry.Key, "/")
-		id, err := strconv.ParseUint(trimmed, 10, 64)
-		if err != nil {
-			return 0, 0, err
-		}
+func getQueueHead(ds datastore.Datastore) (uint64, error) {
```

Comment:
I should document these new methods. I am doing that now.

Reply:
Actually, these methods are internal and relatively small and so I won't add any comments.